pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -203,69 +227,28 @@ func (k *KubeASCheck) runLeaderElection() error {
 	log.Tracef("Current leader: %q, running Kubernetes cluster related checks and collecting events", leaderEngine.GetLeader())
 	return nil
 }
-func (k *KubeASCheck) eventCollectionInit() {
-	if k.latestEventToken == "" {
-		// Initialization: Checking if we previously stored the latestEventToken in a configMap
-		tokenValue, found, err := k.ac.GetTokenFromConfigmap(eventTokenKey, 3600)
-		switch {
-		case err == apiserver.ErrOutdated:
-			k.configMapAvailable = found
-			k.latestEventToken = "0"
-
-		case err == apiserver.ErrNotFound:
-			k.latestEventToken = "0"
-
-		case err == nil:
-			k.configMapAvailable = found
-			k.latestEventToken = tokenValue
 
-		default:
-			log.Warnf("Cannot handle the tokenValue: %q, querying the kube-apiserver cache for events", tokenValue)
-			k.latestEventToken = "0"
-		}
+func (k *KubeASCheck) eventCollectionCheck() (newEvents []*v1.Event, err error) {
+	resVer, lastTime, err := k.ac.GetTokenFromConfigmap(eventTokenKey)
+	if err != nil {
+		return nil, err
 	}
-}
 
-func (k *KubeASCheck) eventCollectionCheck() ([]*v1.Event, []*v1.Event, error) {
-	timeout := time.Duration(k.instance.EventCollectionTimeoutMs) * time.Millisecond
+	timeout := int64(k.instance.EventCollectionTimeoutMs / 1000)
+	limit := int64(k.instance.CardinalityLimit)
+	resync := int64(k.instance.ResyncPeriodEvents)
+	newEvents, k.eventCollection.LastResVer, k.eventCollection.LastTime, err = k.ac.RunEventCollection(resVer, lastTime, timeout, limit, resync, k.ignoredEvents)
 
-	newEvents, modifiedEvents, versionToken, err := k.ac.LatestEvents(k.latestEventToken, timeout)
 	if err != nil {
 		k.Warnf("Could not collect events from the api server: %s", err.Error())
-		return nil, nil, err
-	}
-
-	if versionToken == "0" {
-		// API server cache expired or no recent events to process. Resetting the Resversion token.
-		_, _, versionToken, err = k.ac.LatestEvents("0", timeout)
-		if err != nil {
-			k.Warnf("Could not collect cached events from the api server: %s", err.Error())
-			return nil, nil, err
-		}
-
-		if k.latestEventToken == versionToken {
-			log.Tracef("No new events collected")
-			// No new events but protobuf error was caught. Will retry at next run.
-			return nil, nil, nil
-		}
-		// There was a protobuf error and new events were submitted. Processing them and updating the resVersion.
-		k.latestEventToken = versionToken
-	}
-
-	// We check that the resversion gotten from the API Server is more recent than the one cached in the util.
-	if len(newEvents)+len(modifiedEvents) == 0 {
-		return nil, nil, nil
+		return nil, err
 	}
 
-	k.latestEventToken = versionToken
-	if k.configMapAvailable {
-		configMapErr := k.ac.UpdateTokenInConfigmap(eventTokenKey, versionToken)
-		if configMapErr != nil {
-			k.Warnf("Could not store the LastEventToken in the ConfigMap: %s", configMapErr.Error())
-		}
+	configMapErr := k.ac.UpdateTokenInConfigmap(eventTokenKey, k.eventCollection.LastResVer, k.eventCollection.LastTime)
+	if configMapErr != nil {
```

Comment:
The event collection process relies on the configmap now, so if we can't create/get it we won't collect anything.
I did notice some inconsistencies on the error handling, which I'm updating now though.
We will just have an error log every 15 seconds that says why we can't access the CM (and if eventually they bump the RBAC, it will start working).